Docs/incremental import (#1498)
reimagined/resolve: docs/advanced-techniques.md

## Import-export

Each event store adapter exposes the following API used for event export and import:

| Method | Description                                                           |
| ------ | --------------------------------------------------------------------- |
| export | Returns a readable stream used to export events from an event store.  |
| import | Returns a writeable stream used to import events into an event store. |

In the code sample below, a readable stream returned by an event store's `export` method is pipelined directly into a writable stream returned by a recipient event store's `import` method.

##### Example

```js
import { Readable, pipeline as pipelineC } from 'stream'

import createEventStoreAdapter from 'resolve-eventstore-lite'

const pipeline = promisify(pipelineC)

const eventStore1 = createEventStoreAdapter({
  databaseFile: './data/event-store-1.db'
})

const eventStore2 = createEventStoreAdapter({
  databaseFile: './data/event-store-2.db'
})

await pipeline(eventStore1.export(), eventStore2.import())
```

## Incremental import

> The incremental import allows you to import into an event store only those events that do not already exist in this event store. The incremental import also skips events that are older (have an older timestamp) than the latest event in the recipient event store.

Reviewer: ```suggestion
Incremental import allows you to import into an event store only those events that do not already exist in this event store. Incremental import also skips events that are older (i.e., have an older timestamp) than the latest event in the recipient event store.
```

### Basic Incremental Import

To import events incrementally, pass an array of events to an event store adapter's [incrementalImport](api-reference.md#incrementalimport) method.

> The code sample below demonstrates how to implement and API endpoint that incrementally imports events into the application's event store.

Reviewer: ```suggestion
The code sample below implements an API endpoint that incrementally imports events into the application's event store.
```
